07_VirtualFileSystem/01_README.md

```diff
@@ -0,0 +1,35 @@
+* The external drive file system (probably a vfat, exfat or ntfs file system)
+* The cdrom file system (usually iso9660)
+
+But how can an operating system works with so many different file system, how can they coexist like in linux under the same directory tree? And most important of all: how are we going to implement it?
+
+In this section we are going to understand how everything works, and how the OS can handle so many file systems, we will try to make our basic implementation, to put the knowledge to work, and making our OS able to read and write (kind of...) files. 
+
+It will be divided into two main topics: 
+
+* The Virtual File System (VFS): is a layer that works as an abstraction for the different file systems supported by our os, in this way the application and the kernel use a unified interface of functions to interact with files and directories on different fs, it offers functions like `open, write, opendir`
+* In the File System section we will how they work within a VFS interface, and implement a very basic one to use within our OS,
+
+## A quick recap
+
+Before proceeding is useful to recap some basic file system concepts.
+
+The main purpose of a file system is to store data and make it easily accessible on a human readable way. What a file system does is organize how the data is stored, how they are represented on the disk, and provide functionalities to access, create, update and delete them. More advanced FS can also provide some kind of recovery mechanism (aka `journaling`) but we are not going to cover it because it's out of the scope of this guide. 
+
+Any operating system usually implements it's own FS version, and like many other OS topics, there are different types available that try to solve specific issues, or optimize certain type of operations. 
+
+Any file system is provided by a driver, so an OS has to implement one for each type they want to support (the most popular and easy to implement are probably FAT, tar and ext2).
+
+They will internally represents file, directories in different ways, wether they are just an overhead structure on top of the data, or an entry into an array/list. 
+
+But how to make different file systems with different file representation, be seen in a uniform way by the os? This is achieved through an abstraction layer, the Virtual File System, it will be responsible to talk with the different drivers and provide an unified way to represent them to the user. 
+
+How the different drivers are presented to the user is a design decision, but the two most known (and probably common) way to do that are and of course they have their pros and cons:
+
+* Show them as separated entities, like windows does, where every file system is identified by a unique letter (called the `drive letter`). The pro of this design is that we can have for example same path on two different device with the only difference being the drive letter
```

Comment:
Nice contrast, it could be worth mentioning the multi-root is what NT uses, in contrast to unix which uses the single-root style (as you said below).